src/middleware/authentication/middleware.ts

```diff
@@ -28,5 +47,12 @@ export async function authenticationMiddleware(
     return response.status(401).json({ message: "Token expired" });
   }
 
+  // Store the introspection result in the cache
+  await redisConnector.client.set(
+    accessTokenKey,
+    JSON.stringify(introspectionResult),
+    { EX: cacheExpiry },
```

Comment:
Continued requests with a cached access token within the `cacheExpiry` timeframe will keep it valid up until the original token introspection result has expired.  Is this a behaviour we want or would people prefer we are more aggressive with the cache expiry?

Reply:
I'm thinking we can try 300 (5 min).  This would allow for download of 1000's of responses at a relativley constant latency without having 'hiccups'of potential longer latencies during the process.